# HG changeset patch
# User Max Brister <max@2bass.com>
# Date 1338317462 18000
#      Tue May 29 13:51:02 2012 -0500
# Node ID aebd296a15c4a76166461da006fbd1ffc773deaf
# Parent  8697e3e9d77a2534990578f717e049bbbd04ccf8
Prevent double release of input arguments in JIT

diff --git a/src/pt-jit.h b/src/pt-jit.h
--- a/src/pt-jit.h
+++ b/src/pt-jit.h
@@ -1521,17 +1521,17 @@ private:
   {
     T *ret = new T(arg0, arg1, arg2);
     track_value (ret);
     return ret;
   }
 
   void track_value (jit_value *value)
   {
-    if (value->type ())
+    if (value->type () && ! dynamic_cast<jit_instruction *>(value))
       constants.push_back (value);
     all_values.push_back (value);
   }
 
   // place phi nodes in the current block to merge ref with variables
   // we assume the same number of deffinitions
   void merge (jit_block *merge_block, variable_map& merge_vars,
               jit_block *incomming_block,
